core/test/matrix/hyb.cpp

```diff
@@ -0,0 +1,238 @@
+    {
+        auto v = m->get_const_ell_values();
+        auto c = m->get_const_ell_col_idxs();
+        auto n = m->get_ell_max_nonzeros_per_row();
+        auto p = m->get_ell_stride();
+        ASSERT_EQ(m->get_size(), gko::dim(2, 3));
+        ASSERT_EQ(m->get_ell_num_stored_elements(), 4);
+        ASSERT_EQ(m->get_coo_num_stored_elements(), 1);
+        EXPECT_EQ(n, 2);
+        EXPECT_EQ(p, 2);
+        EXPECT_EQ(c[0], 0);
+        EXPECT_EQ(c[1], 1);
+        EXPECT_EQ(c[2], 1);
+        EXPECT_EQ(c[3], 0);
+        EXPECT_EQ(v[0], 1.0);
+        EXPECT_EQ(v[1], 5.0);
+        EXPECT_EQ(v[2], 3.0);
+        EXPECT_EQ(v[3], 0.0);
+        EXPECT_EQ(m->get_const_coo_values()[0], 2.0);
+        EXPECT_EQ(m->get_const_coo_col_idxs()[0], 2);
+        EXPECT_EQ(m->get_const_coo_row_idxs()[0], 0);
+    }
+
+    void assert_empty(const Mtx *m)
+    {
+        ASSERT_EQ(m->get_size(), gko::dim(0, 0));
+        ASSERT_EQ(m->get_ell_num_stored_elements(), 0);
+        ASSERT_EQ(m->get_const_ell_values(), nullptr);
+        ASSERT_EQ(m->get_const_ell_col_idxs(), nullptr);
+        ASSERT_EQ(m->get_ell_max_nonzeros_per_row(), 0);
+        ASSERT_EQ(m->get_ell_stride(), 0);
+        ASSERT_EQ(m->get_coo_num_stored_elements(), 0);
+        ASSERT_EQ(m->get_const_coo_values(), nullptr);
+        ASSERT_EQ(m->get_const_coo_col_idxs(), nullptr);
+    }
+};
+
+
+TEST_F(Hyb, KnowsItsSize)
+{
+    ASSERT_EQ(mtx->get_size(), gko::dim(2, 3));
+    ASSERT_EQ(mtx->get_ell_num_stored_elements(), 4);
+    ASSERT_EQ(mtx->get_ell_max_nonzeros_per_row(), 2);
+    ASSERT_EQ(mtx->get_ell_stride(), 2);
+    ASSERT_EQ(mtx->get_coo_num_stored_elements(), 1);
+}
+
+
+TEST_F(Hyb, ContainsCorrectData) { assert_equal_to_original_mtx(mtx.get()); }
+
+
+TEST_F(Hyb, CanBeEmpty)
+{
+    auto mtx = Mtx::create(exec);
+
+    assert_empty(mtx.get());
+}
+
+
+TEST_F(Hyb, CanBeCopied)
+{
+    auto copy = Mtx::create(exec);
+
+    copy->copy_from(mtx.get());
+
+    assert_equal_to_original_mtx(mtx.get());
+    mtx->get_ell_values()[1] = 5.0;
+    assert_equal_to_original_mtx(copy.get());
+}
+
+
+TEST_F(Hyb, CanBeMoved)
+{
+    auto copy = Mtx::create(exec);
+
+    copy->copy_from(std::move(mtx));
+
+    assert_equal_to_original_mtx(copy.get());
+}
+
+
+TEST_F(Hyb, CanBeCloned)
+{
+    auto clone = mtx->clone();
+
+    assert_equal_to_original_mtx(mtx.get());
+    mtx->get_ell_values()[1] = 5.0;
+    assert_equal_to_original_mtx(static_cast<Mtx *>(clone.get()));
+}
+
+
+TEST_F(Hyb, CanBeCleared)
+{
+    mtx->clear();
+
+    assert_empty(mtx.get());
+}
+
+
+TEST_F(Hyb, CanBeReadFromMatrixData)
+{
+    auto m = Mtx::create(exec, Mtx::partition::columns, 2);
+    m->read({{2, 3},
+             {{0, 0, 1.0},
+              {0, 1, 3.0},
+              {0, 2, 2.0},
+              {1, 0, 0.0},
+              {1, 1, 5.0},
+              {1, 2, 0.0}}});
+
+    assert_equal_to_original_mtx(m.get());
+}
+
+
+TEST_F(Hyb, CanBeReadFromMatrixDataByPercent80)
+{
```

Comment:
We also need a test for partition::automatically. I know, this does also 80%, but it is still a different execution path we should test.

Reply:
fix it